src/wrapper.rs

```diff
@@ -84,11 +96,14 @@ where
                 .as_ref()
                 .map(|i| (i, &f.ty, FieldKind::from_attrs(&f.attrs)))
         })
-        .filter_map(|(n, t, k)| k.methods(t, n))
+        .filter_map(|(n, t, k)| k.methods(t, n, protobuf))
         .map(|m| m.write_methods(buf))
         .collect::<Result<Vec<_>, _>>()?;
     writeln!(buf, "}}")?;
-    generate_message_trait(&item.ident, prefix, buf)
+    if protobuf {
+        generate_message_trait(&item.ident, prefix, buf)?;
+    }
```

Comment:
Could we pull the call to `generate_message_trait` into the caller and check `protobuf` there?